contracts/strategies/BaseStrategy.sol

```diff
@@ -56,63 +67,31 @@ abstract contract BaseStrategy is IStrategy {
     /// ====================================
 
     /// @notice Initializes the allocation strategy
-    /// @param _allo Address of the Allo contract
     /// @param _identityId Id of the identity
     /// @param _poolId Id of the pool
     /// @param _data The data to be decoded
-    /// @dev This function is called internally by the strategy
-    function __BaseStrategy_init(
-        string memory _strategyIdentifier,
-        address _allo,
-        bytes32 _identityId,
-        uint256 _poolId,
-        bytes memory _data
-    ) internal {
-        if (initialized) {
-            revert BaseStrategy_STRATEGY_ALREADY_INITIALIZED();
-        }
-
-        if (_allo == address(0)) {
-            revert BaseStrategy_INVALID_ADDRESS();
-        }
-
-        initialized = true;
+    /// @dev This function is called by Allo.sol
+    function initialize(bytes32 _identityId, uint256 _poolId, bytes memory _data) external virtual onlyAllo {
+        require(_identityId != bytes32(0), "invalid identity id");
+        require(identityId == bytes32(0), "already initialized");
 
-        _setStrategyIdentifier(_strategyIdentifier);
-
-        allo = Allo(_allo);
         identityId = _identityId;
         poolId = _poolId;
-
-        emit Initialized(_allo, _identityId, _poolId, _data);
-    }
-
-    /// @notice Get the identity id
-    /// @return bytes32 The identity id
-    function getIdentityId() external view returns (bytes32) {
-        return identityId;
-    }
-
-    /// @notice Get the pool id
-    /// @return uint256 The pool id
-    function getPoolId() external view returns (uint256) {
-        return poolId;
     }
 
-    /// @notice Get the Allo address
-    /// @return address The Allo address
-    function getAllo() external view returns (address) {
-        return address(allo);
-    }
```

Comment:
we should keep the getter.
Reason:
Contracts (and Frontends) which interact with the strategies might want to access these variables. Most probably they will only import the Interface contract and not the BaseStrategy contract
Since the variables are defined in the BaseStrategy contract, there is no way to access them using only the Interface.
Yes -> you could query these data also from the Core contract, but it still feels a bit odd to have an interface which does not expose all the main functions and variables.

Reply:
Changed the names to reflect the variable names, made them public variables, and just added getters to the interface to match with this. We can change if you guys prefer the names with `get`.

Reply:
+1 on keeping getter